### PR TITLE
Queue logic

### PR DESCRIPTION
**Pull request by KySo4oK**
Contains only crud for queues without queue_details

### CODE REVIEW

File: backend/src/main/java/com/eshka/controller/QueueController.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    private final QueueService queueService;
+
+    @ApiOperation("get queue by id")
+    @GetMapping("/{id}")
+    public ResponseEntity<QueueResponse> findById(@PathVariable(name = "id") String id) {
+        return new ResponseEntity<>(mapper.queueToQueueResponse(queueService.findById(Long.parseLong(id))),
+                HttpStatus.OK);
+    }
+
+    @ApiOperation("create new subject")
```
Comment (TheBestTvarynka) [inline backend/src/main/java/com/eshka/controller/QueueController.java]: `create new subject` -> `create new queue`

File: backend/src/main/java/com/eshka/controller/QueueController.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    @PostMapping
+    public ResponseEntity<QueueResponse> createSubject(@RequestBody QueueRequest request) {
```
Comment (TheBestTvarynka) [inline backend/src/main/java/com/eshka/controller/QueueController.java]: `createSubject` -> `createQueue`

File: backend/src/main/java/com/eshka/controller/QueueController.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        Queue newQueue = queueService.createNewQueue(mapper.queueRequestToQueue(request));
+        QueueResponse queueResponse = mapper.queueToQueueResponse(newQueue);
+        return new ResponseEntity<>(queueResponse, HttpStatus.CREATED);
+    }
+
+    @ApiOperation("edit queue")
+    @PutMapping
+    public ResponseEntity<QueueResponse> editSubject(@RequestBody QueueRequest request) {
```
Comment (TheBestTvarynka) [inline backend/src/main/java/com/eshka/controller/QueueController.java]: `editSubject` -> `editQueue`

File: backend/src/main/java/com/eshka/controller/QueueController.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        Queue newQueue = queueService.editQueue(mapper.queueRequestToQueue(request));
+        return new ResponseEntity<>(mapper.queueToQueueResponse(newQueue),
+                HttpStatus.OK);
+    }
+
+    @ApiOperation("delete queue by id")
+    @DeleteMapping("/{id}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void deleteSubject(@PathVariable(name = "id") String id) {
```
Comment (TheBestTvarynka) [inline backend/src/main/java/com/eshka/controller/QueueController.java]: `deleteSubject` -> `deleteQueue`

File: backend/src/main/java/com/eshka/service/impl/QueueServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    @Override
+    public Queue editQueue(Queue request) {
+        Queue queue = queueRepository.findById(request.getId()).orElseThrow(
+                () -> new QueueNotFoundException("queue not found"));
+        queue.setDescription(request.getDescription());
+        queue.setTitle(request.getTitle());
+        return queueRepository.save(queue);
+    }
```
Comment (TheBestTvarynka) [inline backend/src/main/java/com/eshka/service/impl/QueueServiceImpl.java]: The logic for editing qeueu is more complex than you think.
Exept title or description, I can change the start and the end date.

**Review by TheBestTvarynka — CHANGES_REQUESTED** [review]
I have a few notes (you can see them in the comments).

File: backend/src/main/java/com/eshka/service/impl/QueueServiceImpl.java (hunk 5 above)
Comment (KySo4oK) [reply backend/src/main/java/com/eshka/service/impl/QueueServiceImpl.java]: > The logic for editing qeueu is more complex than you think.
> Exept title or description, I can change the start and the end date.

but you only about title and description)

**Review by TheBestTvarynka — APPROVED** [review]
:100: :cook_islands:

**Pull request merged**